core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -616,15 +616,21 @@ void maybeProcessFetchQueue() {
                 });
             });
 
-            Set<Object> delayedShareFetchWatchKeys = new HashSet<>();
+            Set<DelayedShareFetchKey> delayedShareFetchWatchKeys = new HashSet<>();
             shareFetchData.partitionMaxBytes().keySet().forEach(
                 topicIdPartition -> {
-                    delayedShareFetchWatchKeys.add(new DelayedShareFetchKey(shareFetchData.groupId(), topicIdPartition));
-                    delayedShareFetchWatchKeys.add(topicIdPartition);
+                    // We add a key corresponding to each share partition in the request so that when there are
```

Comment:
to each share partition in the request => to each share partition in the request in the group